cpp/arcticdb/python/python_to_tensor_frame.cpp

```diff
@@ -73,7 +73,10 @@ NativeTensor obj_to_tensor(PyObject *ptr) {
     auto descr = pybind11::detail::array_descriptor_proxy(arr->descr);
     auto ndim = arr->nd;
     ssize_t size = ndim == 1 ? arr->dimensions[0] : arr->dimensions[0] * arr->dimensions[1];
-    auto val_type = size > 0 ? get_value_type(descr->kind) : ValueType::EMPTY;
+    // In Pandas < 2, empty series dtype is `"float"`, but as of Pandas 2.0, empty series dtype is `"object"`
+    // The Normalizer in Python cast empty `"float"` series to `"object"` so `EMPTY` is used here.
+    // See https://github.com/man-group/ArcticDB/issues/987#issue-1956701712
+    auto val_type = (size == 0 && descr->kind == 'O') ? ValueType::EMPTY : get_value_type(descr->kind);
```

Comment:
Does this mean that if we explicitly set the dtype of an empty column it won't have EMPTY type even though it's empty. I am not sure that this is the desired behavior. This needs unit tests.

Reply:
Yes, this was agreed on https://github.com/man-group/ArcticDB/issues/987 (see the description of the PR).